src/App/Fossa/Analyze.hs

```diff
@@ -232,7 +234,7 @@ analyze (BaseDir basedir) destination override unpackArchives enableVSI filters
   -- This is done because the VSI discover function requires more information than other discover functions do, and only matters for analysis.
   let discoverFuncs' = discoverFuncs ++ [vsiDiscoverFunc enableVSI destination]
 
-  manualSrcUnit <- analyzeFossaDepsYaml basedir
+  (manualSrcUnit, vendoredDeps) <- analyzeFossaDepsYaml basedir
```

Comment:
I think it would substantially simplify the logic here if we:
- move all of the archive upload logic to be called as part of `analyzeFossaDepsYaml` (possibly pass a `Maybe ApiOpts` in there?)
- ignore archives in `--output` mode.

I'd suggest that if we find vendored dependencies in `fossa-deps.yml` in `--output` mode, we'd just log a warning and ignore those dependencies for the scan

Reply:
I strongly agree with your idea to condense the upload logic into part of `analyzeFossaDepsYaml`, but I don't agree with omitting archive deps from the `--output` output. They're easy add and should make debugging projects better. I don't see the benefit of omitting them, other than simplifying the code. Instead of `apiOpts`, we could pass in the `ScanDestination`.

Reply:
I addressed your first point and condensed the archive logic into the yamldeps file. This really cleaned up Analyze. It makes the upload portion of analyze slightly more hidden, but I think it looks cleaner.